File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/locations/City.java
package io.github.tobyhs.weatherweight.api.accuweather.locations;

import com.google.auto.value.AutoValue;
import com.google.gson.annotations.SerializedName;
import me.tatarka.gsonvalue.annotations.GsonBuilder;

/**
 * A city from AccuWeather's Locations API
 */
@AutoValue
public abstract class City {
    /**
     * @return location key
     */
    @SerializedName("Key")
    public abstract String getKey();

    /**
     * @return name of this city
     */
    @SerializedName("LocalizedName")
    public abstract String getLocalizedName();

    /**
     * @return country of this city
     */
    @SerializedName("Country")
    public abstract Country getCountry();

    /**
     * @return administrative area of this city
     */
    @SerializedName("AdministrativeArea")
    public abstract AdministrativeArea getAdministrativeArea();

    @GsonBuilder
    public static Builder builder() {
        return new AutoValue_City.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {
        public abstract Builder setKey(String key);
        public abstract Builder setLocalizedName(String localizedName);
        public abstract Builder setCountry(Country country);
        public abstract Builder setAdministrativeArea(AdministrativeArea administrativeArea);
        public abstract City build();
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/locations/City.kt
package io.github.tobyhs.weatherweight.api.accuweather.locations

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * A city from AccuWeather's Locations API
 *
 * @property key location key
 * @property localizedName name of this city
 * @property country country of this city
 * @property administrativeArea administrative area of this city
 */
@JsonClass(generateAdapter = true)
data class City(
    @Json(name = "Key") val key: String,
    @Json(name = "LocalizedName") val localizedName: String,
    @Json(name = "Country") val country: Country,
    @Json(name = "AdministrativeArea") val administrativeArea: AdministrativeArea,
)
